feeds/wifi-ax/hostapd/patches/f00-019-channel-switch-event.patch

```diff
@@ -0,0 +1,55 @@
+--- a/src/ap/ubus.c
++++ b/src/ap/ubus.c
+@@ -531,6 +531,7 @@ enum {
+ 	CSA_VHT,
+ 	CSA_HE,
+ 	CSA_BLOCK_TX,
++	CSA_REASON,
+ 	__CSA_MAX
+ };
+ 
+@@ -545,6 +546,7 @@ static const struct blobmsg_policy csa_p
+ 	[CSA_VHT] = { "vht", BLOBMSG_TYPE_BOOL },
+ 	[CSA_HE] = { "he", BLOBMSG_TYPE_BOOL },
+ 	[CSA_BLOCK_TX] = { "block_tx", BLOBMSG_TYPE_BOOL },
++	[CSA_REASON] = { "reason", BLOBMSG_TYPE_INT32 },
+ };
+ 
+ #ifdef NEED_AP_MLME
+@@ -994,6 +996,7 @@ hostapd_switch_chan(struct ubus_context
+ 	SET_CSA_SETTING(CSA_VHT, freq_params.vht_enabled, bool);
+ 	SET_CSA_SETTING(CSA_HE, freq_params.he_enabled, bool);
+ 	SET_CSA_SETTING(CSA_BLOCK_TX, block_tx, bool);
++	SET_CSA_SETTING(CSA_REASON, reason, u32);
+ 
+ 	wpa_printf(MSG_INFO, "%s: CSS freq=%d chan=%d sec_chan_off=%d, width=%d, seg0=%d, seg1=%d", __func__,
+ 		css.freq_params.freq,
+@@ -1015,7 +1018,9 @@ hostapd_switch_chan(struct ubus_context
+ 			return UBUS_STATUS_NOT_SUPPORTED;
+ 	}
+ 
+-	hostapd_ubus_handle_channel_switch_event(hapd->iface, HOSTAPD_UBUS_HIGH_INTERFERENCE, css.freq_params.freq);
++	if (css.reason == 0) {
```

Comment:
there are couple of other places this hostapd_ubus_handle_channel_switch_event is called in the hostapd, I saw it in ubus.c and dfs.c . I think we need to add the reason check in the ubus.c where it called from hostapd_dfs_csa_timeout, since that's also triggered by rrm rebalance . please check if we need it for the one in the dfs.c.

Reply:
The call in dfs.c I think can be left along since it seems to only trigger if we detect that radar started transmitting on a DFS channel, so this should only happen automatically and won't be triggered from the cloud. As for the other one I added the same `reason==0` since that seems to be the path we take for a regular channel switch on a DFS channel.